shop/shop/src/main/java/ro/msg/learning/shop/model/Product.java

```diff
@@ -0,0 +1,45 @@
+package ro.msg.learning.shop.model;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import lombok.Data;
+import javax.persistence.*;
+import java.math.BigDecimal;
+import java.util.Set;
+
+@Entity
+@Data
+@Table(name="Product")
+public class Product extends BaseEntity {
+    String name;
+    String description;
+    BigDecimal price;
+    Double weight;
+    @ManyToOne
+    @JoinColumn(name = "categoryid")
+    @JsonIgnore
+    ProductCategory category;
+    @ManyToOne
+    @JoinColumn(name = "supplierid")
+    @JsonIgnore
+    Supplier supplier;
+    String imageUrl;
+    @OneToMany(mappedBy = "product")
+    Set<Stock> stocks;
+
+    @OneToMany(mappedBy = "productorder")
+    Set<OrderDetail> orderDetails;
+
+    public Product() {
```

Comment:
There is a Lombok adnotation for this kind of empty constructor: `@NoArgsConstructor`. There is also `@AllArgsConstructor`